### ISSUE TITLE
Standardized Creative Commons licensing for \setcopyright{rightsretained}

### ISSUE THREAD

**Issue by MatthewFluet** [borisveytsman/acmart#77]
As described at [http://authors.acm.org/main.html](http://authors.acm.org/main.html), authors choosing ACM's author-pays option with non-exclusive permission to publish (which corresponds to executing `\setcopyright{rightsretained}` in an `acmart` document) can also choose to display a Creative Commons License on their works.  Currently, there is not an easy way to do so with `acmart` and certainly not one that easily incorporates into the permission/copyright block.  Perhaps a new `\setcopyright{rightsretained-cc}` or `\setcopyright{creativecommons}` option?

**Comment by MatthewFluet** [comment]
Actually, given the range of Creative Commons Licenses, a single option is probably insufficient.

**Comment by borisveytsman** [comment]
I will contact ACM about this:  each word in the copyright/license part must be approved by their lawyers.

**Comment by MatthewFluet** [comment]
The actual wording of the licensing is prescribed by the Creative Commons; see, for example, https://creativecommons.org/choose/non-web-popup?license_code=by-sa&amp;jurisdiction=&amp;version=4.0&amp;lang=en (but note that there are 6 different CC licenses, induced by two orthogonal choices; see https://creativecommons.org/choose/).

Two points for ACM: their posted author rights say that an author may display a Creative Commons License (so the author should have all 6 choices available to them).  Also, the licensing is a part of the Version of Record, unlike the `authorversion` class option.

**Comment by Craig-Rodkin** [comment]
Hi Matthew and Boris, 

Current method to handle CC licenses is for authors to add the ACM rights text emailed to them into their article. Only after the author pays the APC will the CC license be manually added to their paper. 
Authors are choosing to retain their rights, can free to add the CC license to their own version and post where they wish; only the ACM published version will contain the permission release statement until the APC is paid.

Currently, ACM gives authors 6 months to pay the APC and some are going beyond the 6 months to pay. The only way we can automate the CC license process is if ACM requests payment up front from authors and that decision needs to be made by senior ACM mgt along with the ACM executive committee and council. This potential change will be brought forth soon and I hope it changes so that I don't have to continue to manually add the CC licenses to article PDFs.

FWIW, we have the CC license plug-in added to the ACM Permission Release eRights form. Authors are free to choose whichever license works best for them.

**Comment by borisveytsman** [comment]
Thanks, Craig, 

Should I add the CC plugin to the TeX templates?

**Comment by MatthewFluet** [comment]
I think that the TeX class should support inserting the CC license.  It becomes critical when ACM is using the author's TeX sources to generate the final PDF for the DL.  If the author cannot prepare their document with exactly the permission statement that ACM will ultimately generate, then they cannot properly handle column/page breaks and figure placements.

Craig, I perfectly understand that ACM needs to have a process in place to coordinate ACM eRights, but it is somewhat independent of the class file.

**Comment by Craig-Rodkin** [comment]
Hi Matthew,

I do not reprocess the formatted file, but rather remove the permissions block and adjust the copyright statement, then add the CC license and text (with a link to the actual CC License page). Since I only add a layer to the PDF, the integrity of the author’s formatting remains intact.

We are certainly on the same page with integrating CC licenses into the template. ACM just needs to shift APC payment to the beginning of the process rather that post publication. I am putting together a proposal to the Director of Publications this week to make the change. He will then move the proposal up the food chain.

Regards,

-Craig

Craig Rodkin
Publications Operations Manager
Association for Computing Machinery
rodkin@hq.acm.org<mailto:rodkin@hq.acm.org>
http://member.acm.org/~rodkin

Explore Association for Computing Machinery's AUTHOR-IZER SERVICE:  http://www.acm.org/publications/acm-author-izer-service<http://www.acm.org/publications/acm-author-izer-service>

**Comment by Craig-Rodkin** [comment]
Boris, Not yet.

It only makes sense if we can move the APC payment upfront. Otherwise, I would have to remove the CC licensees for all unpaid APCs. Also, since the CC license icon and text take up much less space, I would mean that I would have to muck with the text layout on the first page in order to fit in the ACM rights text.
Moving in this direction would require additional manual work which, right now, would be a step backwards.

Regards,

-Craig

Craig Rodkin
Publications Operations Manager
Association for Computing Machinery
rodkin@hq.acm.org<mailto:rodkin@hq.acm.org>
http://member.acm.org/~rodkin

Explore Association for Computing Machinery's AUTHOR-IZER SERVICE:  http://www.acm.org/publications/acm-author-izer-service<http://www.acm.org/publications/acm-author-izer-service>

**Comment by MatthewFluet** [comment]
Craig: note that the new class file does not use a fixed-size permissions block (see #10); it would seem that adjusting the permissions block relative to what the author submitted will either overflow into the footer or leave extra white space.

**Comment by Craig-Rodkin** [comment]
Matthew, when I manually add the CC icon and text, there is a little extra whitespace, but never any footer overflow (unless the author circumvents the page specifications).

I’ve been replacing the rights text with the CC license icon and text for almost a year now and have yet to have any complaints from authors using any of the previous templates.  I wish to continue this practice until a formal decision on APC payment timing is made.

Believe me, I’m not looking to do the additional manual work, but I the alternative manual intervention of removing a CC icon and text and adding the ACM permissions text will definitely lead to mucking with the first page formatting and take longer to change.

Regards,

-Craig

Craig Rodkin
Publications Operations Manager
Association for Computing Machinery
rodkin@hq.acm.org<mailto:rodkin@hq.acm.org>
http://member.acm.org/~rodkin

Explore Association for Computing Machinery's AUTHOR-IZER SERVICE:  http://www.acm.org/publications/acm-author-izer-service<http://www.acm.org/publications/acm-author-izer-service>

**Comment by dbeyer** [comment]
@MatthewFluet: Why is it not sufficient to support only one version of CC?
Is @Craig-Rodkin really adding different versions of CC to the papers, or only the one that ACM currently agreed on?

**Comment by dbeyer** [comment]
@Craig-Rodkin, is there an outcome of your proposal already?
Many conferences require the authors to pay a full conference-registration fee *before* the camera-ready deadline, and papers without that are removed from the proceedings.
I would see no problem with requiring payment of the APC before the camera-ready deadline (or up to two weeks later), and if there is no payment, then there is no open access.

@Craig-Rodkin, I would like to understand your argument about the dependency:
You say that you can add the CC text only after the APC is paid because otherwise you need to change the license text.
But don't you need to change the license text anyway if an author does not pay?
In other words, if the APC is not paid, then the license switches from the non-exclusive remission release to the exclusive license.  Or do you leave those papers with the permission release although the fee is not paid?

**Comment by MatthewFluet** [comment]
@dbeyer Regarding "Why is it not sufficient to support only one version of CC?": Well, there are actually 6 different CC licenses (not counting older versions).  I thought that ACM left the choice of which license up to the author when paying for open access, but maybe I misremembered  that conversation.

**Comment by Craig-Rodkin** [comment]
Hi,

There are actually 7 CC licenses (everyone forgets CC-0; public domain) an author gets to choose from. ACM cannot and will not dictate which rights option an author can shoose.

Regards,

-Craig

Craig Rodkin
Publications Operations Manager
Association for Computing Machinery
rodkin@hq.acm.org<mailto:rodkin@hq.acm.org>
http://member.acm.org/~rodkin

Explore Association for Computing Machinery's AUTHOR-IZER SERVICE:  http://www.acm.org/publications/acm-author-izer-service<http://www.acm.org/publications/acm-author-izer-service>

**Comment by Craig-Rodkin** [comment]
APC’s are and always will be tied to rights, so the business decision was to add the APC inquiry to the completion of the rights form. There are other logistic reasons for authors not to provide APC dollars at the time of registration as well, but will not discuss this here.

>>But don't you need to change the license text anyway if an author does not pay?
In other words, if the APC is not paid, then the license switches from the non-exclusive remission release to the exclusive license

Yes, I manually change the rights information on papers.

Regards,

-Craig

Craig Rodkin
Publications Operations Manager
Association for Computing Machinery
rodkin@hq.acm.org<mailto:rodkin@hq.acm.org>
http://member.acm.org/~rodkin

Explore Association for Computing Machinery's AUTHOR-IZER SERVICE:  http://www.acm.org/publications/acm-author-izer-service<http://www.acm.org/publications/acm-author-izer-service>

**Comment by borisveytsman** [comment]
@Craig-Rodkin I can easily add all seven CC licenses allowing the author to choose among them

**Comment by Craig-Rodkin** [comment]
They shouldn’t be part of the template…at least for now.

I’ve had many internal discussions about this and we’ve all come to the same conclusion.

Regards,

-Craig

Craig Rodkin
Publications Operations Manager
Association for Computing Machinery
rodkin@hq.acm.org<mailto:rodkin@hq.acm.org>
http://member.acm.org/~rodkin

Explore Association for Computing Machinery's AUTHOR-IZER SERVICE:  http://www.acm.org/publications/acm-author-izer-service<http://www.acm.org/publications/acm-author-izer-service>

**Comment by dbeyer** [comment]
@Craig-Rodkin what does it take to convince you to let Boris include the CC-BY in the template?

There is a gold-open-access journal (PACMPL), for which the APC is guaranteed to be paid.
They editorial board just recommended on their web page that the board recommends the authors to use CC-BY, which means that one would expect the majority of articles using CC-BY.

The above-mentioned checking of payments and possible removal of CC-BY text from papers that are late with the payments does not apply to the >40 PACMPL papers in the first issue,
and all future issues as far as I understood.

Not letting the authors add the CC-BY text via the ACM template creates extra work for ACM (you need to fix ~40 papers by adding the CC-BY text) and for the authors (they want to add the CC-BY text immediately to their own versions).

I personally have no opinion on this, but heard from many authors that they would really like to have the option in the template. It is just an option in the end.

If the point is that ACM wants to avoid authors to prematurely use the option, it is up to the author instructions of the conference / journal to tell the authors to not use it,
and up to the processing vendor to check that the authors adhere.

Or did I misunderstand the topic?

Cheers,
Dirk

**Comment by Craig-Rodkin** [comment]
Dirk,

ACM will add the CC licenses on the papers before they are published to the DL. ACM is not altering the templates at this point in time. I really do not want to keep addressing this point as it takes away from other deadline work.

When camera ready copy is ready for the PACMPL articles, please send them to Anna and myself for updating.

Regards,

-Craig

Craig Rodkin
Publications Operations Manager
Association for Computing Machinery
rodkin@hq.acm.org<mailto:rodkin@hq.acm.org>
http://member.acm.org/~rodkin

Explore Association for Computing Machinery's AUTHOR-IZER SERVICE:  http://www.acm.org/publications/acm-author-izer-service<http://www.acm.org/publications/acm-author-izer-service>

**Comment by xuhdev** [comment]
I understand that ACM has decided to not add this option for CC licenses to the template. Is it possible, though, to add an option to use a user-custom permission text, (e.g., `\acmcopyrighttext`) just to ease the release of author's version?

**Comment by wadler** [comment]
Craig, I would like to add my voice to those of Matthew, Dirk, and Harry to urge that the acmart template be modified to make it easy for authors to include a CC copyright notice. ACM should be supporting researchers in publishing their research and getting it used. Consider an author preparing a preprint and circulating it for comment: they would be greatly helped by being able to label it CC-BY, thereby making it clear they are happy for others to build on their research but that they want to be given credit when quoted.

I take the point that ACM is concerned about getting paid for journals, unlike PACMPL, where it is the author not the SIG that is responsible for fees. But that should not be a reason to make it harder for authors, and for those who prepare ACM publications, to prepare publications with a CC license. I know there has been a lot of back and forth before, but I would still appreciate a response.

**Comment by samth** [comment]
I recently submitted a paper for which I paid the APC and chose a CC licence. I am unable to format the paper using acmart.cls with an accurate copyright and license block, because the class file does not have an accurate way to specify it. I note that some conferences have exceptions built in to the class file which generate CC licenses, and this feature should be added generally.

**Comment by krono** [comment]
I also would support being able to specify CC licenses.

Having dealt with that, it turns out that the CC iconography also makes it more immediately clear _what_ rights authors have granted, which takes much longer for non-CC licensed stuff.

**Comment by jilltxt** [comment]
I have also prepaid the APC and chosen a CC license for my ACM conference paper. The process seems very unclear - the email asks me to accept the "final" PDF with an incorrect copyright block. This doesn't seem reasonable given that I have prepaid the APC. I have been assured by email that the copyright will be changed before upload to the DL, but nobody can tell me what the final copyright block will say. This seems like a very strange way to organize things.

**Comment by dbeyer** [comment]
Hi Jill,

There is a clear process, but the process is perhaps not known to many authors:
At the time the proceedings-processing vendor finalizes the proceedings, the APC charges are often not yet processed. Therefore, the process of replacing the copyright text is not handled by the (ACM-external) vendor, but internally at the ACM headquarter.
Sometimes the payment takes several months, sometimes authors decide later to actually not pay,
therefore, it has to be done after the payment is confirmed.

Thus, when you are asked to approve your final paper, this is meant "modulo" the publishing-rights text. Later, ACM will touch your PDF and replace the legal wording.

> nobody can tell me what the final copyright block will say

In case you want to see an example: https://doi.org/10.1145/3290314
(This is an article in a journal, thus it uses a different style, but you get the idea.
 Also, this is an open-access journal and thus, the vendor actually does place the CC-BY logo
 before delivery to ACM DL, because SIGPLAN guarantees that the APC will be paid.)

Cheers,
Dirk

**Comment by Craig-Rodkin** [comment]
Hi everyone. 

Thanks for responding Dirk. 

There is another reason why this must be ACM's workflow.

Once an article is published in the ACM Digital Library with a CC license, ACM cannot legally remove it from the article as we too are governed by the rights the author has chosen on the ACM eRights form to publish their article. Therefore we collect the APCs before we add the CC license to an article.

As Dirk has pointed out, there are authors who choose the OA APC option, but never pay. Unfortunately, there is fairly high percentage of authors who do this. For some it may be a language barrier, for others, they are looking to game the system .

Please note, that for those authors who choose the OA APC CC option on the ACM eRights form, we provide the CC license information to the authors on their completed rights form. This way, authors can add the CC license information to their articles to post on their institutional, personal, or mandatory repository.

**Comment by jilltxt** [comment]
Thanks for the example - but I prepaid the APF two weeks ago, using a credit card at the time I submitted the information. Surely it should be possible to have a different workflow when the author has already paid?

**Comment by Craig-Rodkin** [comment]
Yes it is possible and we are working towards that goal with the implementation of the new template workflow and production system integration (https://www.acm.org/publications/taps/word-template-workflow). Unfortunately, we are just not there yet.

**Comment by thomwiggers** [comment]
It's been three years, and it would be helpful for full versions of (OA) papers to allow the authors to include the right license without hacking the `acmart.cls` file.  For example, we have a paper where appendices exceed the page limit, forcing us to moves parts of a formal proof statement outside of the ACM version. Although we will make available the proof as auxiliary material published along it on the ACM site, it's nice if the version on eprint will have the proof in its proper place...

**Comment by Craig-Rodkin** [comment]
Hi Thom,

Unfortunately, we only add the Creative commons license after payment has been received. Either Staff or our vendors add the statement. 

As we move forward with production using The ACM Production System (TAPS), the generation of output will be left to the system to complete and as long as the APC is paid before production processing is complete, the correct rights information will be added to the published work. Of course authors have the chance to review and approve output, providing authors a chance to work with support to make layout changes.

**Comment by samth** [comment]
@Craig-Rodkin I just want to note here how incredibly disappointing it is, as a member of the ACM and a regular author of ACM publications, that ACM policy is to explicitly make our tools less useful so as to preserve ACM's control over how license information is presented in other places.

**Comment by JonathanAldrich** [comment]
@Craig-Rodkin I agree with Sam, Thom, and others--we should fix this. It's a completely reasonable use case that's being discussed here.  It doesn't make sense to try to use a .cls file to enforce ACM policy; that unnecessarily limits authors who want to prepare their own versions of material.  Instead, we should enforce ACM policy in the obvious way, by making sure the license in the PDF we get matches the one the author arranged for.

**Comment by thomwiggers** [comment]
Whatever the ACM does with "the published work" is fine, but this template is also used to prepare versions that are *not* going to go through the ACMs systems and this template has explicit support for that: for example, the `authorversion` options. It would not make sense to upload a version that has `\setcopyright{rightsreserved}` on my own website, while the ACM DL version has `CC-BY`.

**Comment by dvanhorn** [comment]
As a long time author of ACM published materials who does a couple hundred hours of volunteer review work for ACM publication venues each year, I want to second @samth and @JonathanAldrich here (and the many others who spoke up earlier).  Deliberately making tools worse to extract rent is not the way a non-profit professional organization should be run.  @Craig-Rodkin, if the ACM wants to make sure license info matches what the author arranged for, then the ACM should do its job and check it.

**Comment by rionda** [comment]
Totally agree with the above. Since the Authors retain the rights to publish a version on their website, and the class supports this possibility through the`authorversion` option, there must be a way to specify the correct license from the class without much trouble.

**Comment by borisveytsman** [comment]
I am fine with adding this option if ACM agrees

**Comment by leonmoonen** [comment]
I'm running in this issue as well, where I have paid the open access fees and indicated I want to publish under CC-BY 4.0, but am still forced to submit a camera-ready version of the paper with a \setcopyright{rightsretained} blurb that is incompatible with the European Plan S as well as my Norwegian funding agency (RCN):  the \setcopyright{rightsretained} blurb prohibits that copies are made or distributed for profit or commercial advantage but Plan S and RCN explicitly forbids the CC-BY-NC license on grounds that the No Commercial restriction hinders innovation potential. There are several examples in the Digital Library where the blurb was kept on the paper and just a CC-BY line/logo was added, making the paper incompatible with Plan S / RCN requirements. 

If authors are supposed to prepare a camera-ready version, no additional replacement magic should be needed (esp., since it clearly can go wrong). Having this text as a warranty for authors who do not pay the OA fees does not make sense, such papers can simply be kept/removed from the Digital Library, as is done for authors that do not register for a conference.

**Comment by Craig-Rodkin** [comment]
>>). Having this text as a warranty for authors who do not pay the OA fees does not make sense, such papers can simply be kept/removed from the Digital Library, 

ACM enters into a binding contract with the author when they complete their rights form. As per the agreement and the conditions of CC licenses, once we publish an article with a CC license, ACM cannot simply remove the CC license or delete the article from the Digital Library. Also, ACM Publications Policies do not allow for the removal of articles from the Digital Library for non-payment of OA APCs. To read more about ACM publication policies, please visit https://www.acm.org/publications/policies/

Should you have an issue where you paid the OA APC, your paper is published in the Digital Library, and the CC License you chose is not being displayed on the paper, please contact rightsreview@acm.org for resolution. If your article in still in the production process, please let me know the publication title and your article title, the DOI assigned and we will make sure that the OA is added. Thank you.

**Comment by rionda** [comment]
"There are several examples in the Digital Library where the blurb was kept on the paper and just a CC-BY line/logo was added, making the paper incompatible with Plan S / RCN requirements." ->  Aren't the blurb created by "rightsretained" and the rules sets by CC-BY actually incompatible? Having both makes no sense, IMHO.

**Comment by Craig-Rodkin** [comment]
Please send me us the list of papers. If a CC-BY was added, the permissions statement must be removed. Please send the information to ***@***.******@***.***> for resolution.

For CC-BY articles, we make the changes on the back end during production, so the data added to the source file is overwritten,  or make the change post-publishing when payment is sent after the article is available in the DL . I will need the list of published article DOIs where the CC license is appearing and the rights statement was published in error. Once we have this information we will investigate where the problem is in the workflow. thanks

Be Well
Regards,

Craig

Craig Rodkin
ACM Publications Operations Manager
***@***.******@***.***>

**Comment by leonmoonen** [comment]
Unfortunately, GitHub masked the email address, but two examples (not my papers) are https://dl.acm.org/doi/pdf/10.1145/3083671.3083678 and https://dl.acm.org/doi/pdf/10.1145/3328320.3328385. You can find more using a full-text search on a part of the "blurb" in combination with CC-BY terms (which does give a bunch of false positives as well so it's kind of tedious to create such a list. However, there may be internal DL tools that allow restricting the search to the first page?)

**Comment by leonmoonen** [comment]
Moreover, for papers that *do* show a CC-BY license, the old license text is not removed from the PDF, it is just being superimposed with a picture but it can still be selected. For example https://dl.acm.org/doi/pdf/10.1145/3405656.3420232
![Screen Shot 2021-07-30 at 15 51 03](https://user-images.githubusercontent.com/2410711/127663343-5e228adc-d655-4328-a7fd-7dd566d10ee9.png)
I feel having both there raises questions on what is the actual license, and it would be far more preferable if authors can indicate the correct CC-BY license in the camera-ready.

**Comment by Craig-Rodkin** [comment]
I took a look at https://dl.acm.org/doi/pdf/10.1145/3083671.3083678, the author requested the OA option, but did not pay the APC to date. The PDF is correct.

The https://dl.acm.org/doi/pdf/10.1145/3328320.3328385 PDF incorrectly contains the ACM permission along with the CC license. We will correct this ASAP.

We will look into the best way to identify this problem and rectify papers which contain the incompatible rights information.

I would like to thank you again for alerting us to this problem
Be well.
Regards,

Craig

Craig Rodkin
ACM Publications Operations Manager
***@***.******@***.***>

**Comment by rionda** [comment]
Referring to the first PDF, how can a work be, at the same time, under CC-BY, but also one cannot make copies for profit or for commmercial work? That does not sound solid from a legal point of view.

**Comment by Craig-Rodkin** [comment]
Thanks for this example as well. We will work on a solution.

ACM will revisit the availability of the CC-BY in the templates after the ACM Digital Library converts to being completely open.

Be well.
Regards,

Craig

Craig Rodkin
ACM Publications Operations Manager
***@***.******@***.***>

**Comment by Craig-Rodkin** [comment]
It’s incompatible. Which is why we need to go back and correct the papers (remove the permission statement) when a CC-BY license is published. We are working on this and will get all articles corrected ASAP.

Be well.
Regards,

Craig

Craig Rodkin
ACM Publications Operations Manager
***@***.******@***.***>

**Comment by leonmoonen** [comment]
Hi Craig,

> ACM will revisit the availability of the CC-BY in the templates after the ACM Digital Library converts to being completely open.

Out of interest, can you share (a pointer to) the rough timeline for that?

**Comment by Craig-Rodkin** [comment]
Sure. Please see https://www.acm.org/articles/pubs-newsletter/2020/blue-diamond-october2020

Be well.
Regards,

Craig

Craig Rodkin
ACM Publications Operations Manager
***@***.******@***.***>

**Comment by borisveytsman** [comment]
I can only add, that as soon as ACM approves, I will add the licenses.

**Comment by rionda** [comment]
It is unclear to me why it is necessary to wait for ACM approval. The features supported by this template should be decoupled from ACM's publication process. 

For example, at this point, there is no way for an author of a CC-BY paper to create an `authorversion` of the work to post on their website, because the template does not support specifying that the work is under CC-BY. 

Note that it is already possible for an author to submit a camera-ready version that is not in line with what they selected on the copyright form (e.g., they may have assigned the copyright to ACM in the form, but use `acmlicensed` or `rightsretained` or other options), so in no way the template is enforcing this kind of constraints or preventing mistakes (nor it should, IMHO).

**Comment by borisveytsman** [comment]
My understanding is that ACM owns the style and has the ultimate say about its features.

**Comment by rionda** [comment]
This style is distributed under the LaTeX Project Public License. I'm not a lawyer, but ACM does not own it according to any definition of the word *own*.  It's not even clear that ACM has *complete* copyright on it: multiple people contributed to this style, and at least some certainly did not sign a release of their copyright to ACM.

You can decide to not implement features until you have approval from ACM, but that's different. 

Personally, I don't think it is good service to wait for this approval in this specific case: as mentioned above, there are *legit* and *ACM-sanctioned* scenarios that are not supported by the style, and I would say they should.

**Comment by Craig-Rodkin** [comment]
Hi Matteo,

ACM has hired Boris as a consultant to create a template which is compatible with ACM’s production systems and adheres to ACM’s business policies. The template is on the GitHub site to correct bugs, and provide support for authors who are publishing with ACM and is compatible with current production practices.

ACM has instructed Boris to add features to the template over the years which do help authors (outside of ACM’s publishing production processes) as a good work of the ACM, but they are not incompatible with the ACM production system or in conflict with current business policies. As I’ve previously stated on this and I think a few other GitHub threads, once ACM Digital Library content is freely available to all users, the ACM Publications and ACM Council will need to revisit this matter.


Be well.
Regards,

Craig

Craig Rodkin
ACM Publications Operations Manager
***@***.******@***.***>

**Comment by rionda** [comment]
Hi Craig,

There exists scenarios where the Authors would like to use the template to perform actions that are sanctioned by ACM (posting the final version on one's personal/company website, which is always allowed, but I'm referring specifically to the case of a work having a CC-BY licens) but the template currently does not support creating a version for this scenario.

It would be *very* "good work of the ACM", and *really* "help authors" to make it possible. 

Adding these features to support these use cases does not seem to be in any way incompatible with the current production process or business policies.

ACM should really embrace and *welcome* the fact that a *community* has developed around this style, with people suggesting ideas, enhancements, and reporting bugs. It should be seen as a huge success of an *open* initiative, and more participation should be encouraged.

Thanks,
Matteo

**Comment by JonathanAldrich** [comment]
Quick note to Matteo--I'm concerned about this issue too (see my note in the chain above) and I think we may be able to discuss this in some publications board body soon.  So stay tuned.

I do want to correct one misconception in your latest message.  Authors *do* have the right to post an author-prepared version of their manuscript to their personal website, but not the final version of record--that is only for the ACM Digital Library, in the case where a CC-BY license is not used.  Of course, in the case of CC-BY, sharing the final version anywhere is permitted.  For full information, see https://www.acm.org/publications/openaccess

Best,

Jonathan

**Comment by borisveytsman** [comment]
I agree with @JonathanAldrich that the proper venue for this discussion is not github issue tracker.  At the end of the day it is the ACM members and SIGs that should voice their preference and tell ACM publication authorities what they want.

My choice would be an interface that allows the author to attach any license to the `authorversion` with certain texts prefilled by ACM.  However, I do not feel I have the right to do it right now.

If there is no objection, I will close this issue since it is not the best forum for this discussion.

**Comment by rionda** [comment]
Yes, I misused the word "final" in my post.

Thanks for considering this use case

I'm fine with closing the issue.

**Comment by jackbergus** [comment]
If this helps, I was told that using _acmengage_ should also activate the possibility of setting up the CC licence, albeit the font size might slightly increase.
